samples/minimum_criteria.rst

```diff
@@ -0,0 +1,98 @@
+++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added.
+    Testcases are mandatory, and having samples may be optional.
+  * The primary purpose of a sample is to provide a reference to the user.
+  * Samples must not use Zephyr's testing framework.
+
+    * Must not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+    * Must not use zasserts or asserts in samples.
+
+2. Samples should target multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
+    of testing to the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of ``integration_platforms`` to limit the scope when there are timing or
+    resource constraints.
+  * Make the sample as generic as possible. Avoid making a sample platform specific unless it is
+    for particular hardware.
+
+  .. _Testcase Docs: https://docs.zephyrproject.org/latest/develop/test/twister.html?highlight=test%20runner#test-cases.
+
+3. Twister should be able to build every sample.
+++++++++++++++++++++++++++++++++++++++++++++++++
+  * Every sample must have a YAML file.
+
+    **For example:**
+
+    .. code-block:: yaml
+
+      tests:
+        sample.kernel.cond_var:
+          integration_platforms:
+            - native_posix
+          tags: kernel condition_variables
+          harness: console
+          harness_config:
+            type: one_line
+            regex:
+              - ".*Waited and joined with 3 threads"
+
+  **Guidelines for Samples:**
+    * Minimize the use of ``platform_allow`` and architecture filters.
+    * Do not mark the test as build only unless necessary.
+    * Any test case-specific configuration options are added using ``extra_args`` or
+      ``extra_configs`` options in the YAML file. The prj.conf file should have the in-common
+      configuration options.
+    * There can be multiple testcases in the sample's YAML file to verify if the feature works
+      with different configurations. For example, different logging modes are tested using
+      multiple test scenarios in samples/subsys/logging/syst.
+    * Sample output can be validated leveraging the ``harness_config`` regex option,
+      wherever applicable.
+
+4. A sample should provide sufficient coverage of a subsystem, feature, or module.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  **DO's:**
+    * Samples should provide integration testing.
+    * Cover the most common and important use cases of the functionality.
+    * Keep the code simple and easy to read. Example: samples/philosophers.
+
+  **DONT's:**
+    * Samples must not test the negative or edge case behaviors.
+    * Must not be unit tests.
+    * Must not incorporate more than one subsystem or feature.
```

Comment:
We state that samples aren't tests early on, but keep using test terminology to describe them?

Reply:
agree.

>  * Samples should provide integration testing.

This will be confusing, we should be testing the sample functionality itself, not what it integrates and the features it has, this is just to make sure that samples are still operational. Integration testing is more than that, from wikipedia:

> Integration testing (sometimes called integration and testing, abbreviated I&T) is the phase in [software testing](https://en.wikipedia.org/wiki/Software_testing) in which individual software modules are combined and tested as a group.